python/vyos/ifconfig/bridge.py

```diff
@@ -274,20 +274,31 @@ def update(self, config):
         for member in (tmp or []):
             if member in interfaces():
                 self.del_port(member)
-        vlan_filter = 0
 
-        vlan_del = set()
-        vlan_add = set()
+        # enable/disable Vlan Filter
+        vlan_filter = 1 if 'enable_vlan' in config else 0
+        self.set_vlan_filter(str(vlan_filter))
+
+        if vlan_filter:
+            del_ifname_vlan_ids = get_vlan_ids(ifname)
+            # Delete all VLAN IDS configured in the interface first
```

Comment:
Is it wise to always delete all VLANs first and then re-add them? This heavily sounds like traffic interruption which should be avoided at any cost.

See the example about IP addresses when they are added to the system. I first retrieve the current list of assigned addresses, calculate the difference and only remove those who are not longer present in the configuration.